warehouse/query-core/src/test/java/datawave/query/planner/rules/NodeTransformVisitorTest.java

```diff
@@ -0,0 +1,185 @@
+package datawave.query.planner.rules;
+
+import datawave.query.config.ShardQueryConfiguration;
+import datawave.query.jexl.JexlASTHelper;
+import datawave.query.jexl.nodes.QueryPropertyMarker;
+import datawave.query.jexl.visitors.JexlStringBuildingVisitor;
+import org.apache.commons.jexl2.parser.ASTAndNode;
+import org.apache.commons.jexl2.parser.ASTJexlScript;
+import org.apache.commons.jexl2.parser.JexlNode;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static org.apache.commons.jexl2.parser.JexlNodes.children;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public class NodeTransformVisitorTest {
+    
+    private static final List<String> PATTERNS = Arrays.asList(new String[] {".\\.\\*", "\\.\\*.", "\\.\\*<[^<>]+>"});
+    private static final RegexPushdownTransformRule regexPushdownRule = new RegexPushdownTransformRule();
+    private static final NodeTransformRule reverseAndRule = new NodeTransformRule() {
+        @Override
+        public JexlNode apply(JexlNode node, ShardQueryConfiguration config) {
+            if (node instanceof ASTAndNode) {
+                // reverse the children
+                ArrayList<JexlNode> children = newArrayList();
+                for (JexlNode child : children(node)) {
+                    children.add(child);
+                }
+                Collections.reverse(children);
+                return children(node, children.toArray(new JexlNode[children.size()]));
```

Comment:
Doesn't matter since this is a test, but the current guidance I've seen is that creating an array of size 0 is actually faster (and safer--no race condition) that making an array of the list size. That is, this is preferred:
```java
return children(node, children.toArray(new JexlNode[0]));
```
[Here's](https://shipilev.net/blog/2016/arrays-wisdom-ancients/) a good article on the subject.